internal/productcore/disable.go

```diff
@@ -0,0 +1,57 @@
+package productcore
+
+import (
+	"io"
+
+	fsterr "github.com/fastly/cli/pkg/errors"
+
+	"github.com/fastly/cli/pkg/argparser"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// Disable is a base type for all 'disable' commands.
+type Disable[O any] struct {
+	Base
+	hooks *EnablementHookFuncs[O]
```

Comment:
I'm not sure of any reason that `hooks` needs to be of pointer type. Should be fine to use the struct's value here and in the other generic action implementations.

Reply:
During testing the 'global' EnablementHookFuncs structure in each package gets modified by the tests, and so if a copy was made during the instantiation of this structure (which happens during the CLI startup) the mock functions would not be called.

Reply:
I'll add a comment to the source during the next round of edits, so the next person with this question will have an answer!